proto/asn1-pdu/common.h

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef PROTO_ASN1_COMMON_H_
+#define PROTO_ASN1_COMMON_H_
+
+#include <math.h>
+#include <stdint.h>
+
+#include <vector>
+
+constexpr uint8_t kAsn1Constructed = 0x20u;
+constexpr uint8_t kAsn1Universal = 0u;
+constexpr uint8_t kAsn1Application = 0x40u;
+constexpr uint8_t kAsn1ContextSpecific = 0x80u;
+constexpr uint8_t kAsn1Private = 0xc0u;
+
+// Integer is UNIVERSAL 2 (X.680 (2015), 8.6, Table 1) and is always primitive
+// (X.690 (2015), 8.3.1).
+constexpr uint8_t kAsn1Integer = kAsn1Universal | 0x2u;
+// Bitstring is UNIVERSAL 3 (X.680 (2015), 8.6, Table 1) and is always
+// primitive in DER (X.690 (2015), 10.2).
+constexpr uint8_t kAsn1Bitstring = kAsn1Universal | 0x03u;
+// UTCTime has tag number 23 (X.680 (2015), 8.6, Table 1) and is always
+// primitive in DER encoding (A Layman's Guide to a Subset of ASN.1, BER, and
+// DER, 5.17).
```

Comment:
In general, I want to avoid quoting the Layman's Guide, because that's a secondary reference, not a primary reference :)

```suggestion
// primitive in DER encoding (X.690 (2015), 10.2).
```

Context (not for this file, just background for your own education :D ):
The reason it's 10.2 is that UTCTime is actually just a fancy "useful alias" to
`UTCTime ::= [UNIVERSAL 23] IMPLICIT VisibleString` that was added later during standardization (see J.1 of X.680). `VisibleString` is in Table 8 of X.680 (2015), 4.1, and so the 10.2 clause from X.690 applies.